Lib/test/test_typing.py

```diff
@@ -8484,6 +8484,41 @@ class X(Annotated[int, (1, 10)]): ...
         self.assertEqual(X.__mro__, (X, int, object),
                          "Annotated should be transparent.")
 
+        class Y1(Annotated[list[int], 'meta']): ...
+        self.assertEqual(Y1.__mro__, (Y1, list, object))
+
+        class Y2(Annotated[List[int], 'meta']): ...
+        self.assertEqual(Y2.__mro__, (Y2, list, Generic, object))
+
+        class Y3(
+            Annotated[
+                Annotated[
+                    list[
+                        Annotated[
+                            list[
+                                Annotated[int, 'm'],
+                            ],
+                            'm',
+                        ],
+                    ],
+                    'm',
+                ],
+                'm',
+            ],
+        ): ...
+        self.assertEqual(Y3.__mro__, (Y3, list, object))
+
+        T = TypeVar('T')
+        class GenericBase(Generic[T]): ...
+        class Mixin: ...
+        class Z(Annotated[GenericBase[int], 'meta'], Mixin): ...
+        self.assertEqual(Z.__mro__, (Z, GenericBase, Generic, Mixin, object))
+
+        # Errors:
+        A1 = Annotated[Literal[1], 'meta']
```

Comment:
Can we add an error case with a SpecialForm that is not a GenericAlias, like NoReturn or Self?